app/controllers/admin/questions_controller.rb

```diff
@@ -46,15 +47,10 @@ def update
 
   def move
     target = Question.find(params[:target])
-    method = case params[:relation]
-      when 'before' then :prepend_sibling
-      when 'after' then :append_sibling
-      when 'inside' then :prepend_child
-    end
-
-    target.send(method, @question)
+    QuestionMover.new(current_division: selected_division, question: @question,
+                      target: target, relation: params[:relation].to_sym).move
     render_set_json(@question.question_set)
-  rescue
+  rescue ArgumentError
     flash.now[:error] = I18n.t('questions.move_error') + ": " + $!.to_s
```

Comment:
we aren't expecting users to see this error right, it's just a fallback if something in the browser fails?

Reply:
It does actually show up on the page due to some pre-existing error display machinery. But the user likely won't see it b/c it shows at top of the page and user is almost always scrolled down. But it could be helpful for helping us diagnose. You are right that it would be unexpected for that error to show up and would mean we have a bug.